x/clock/genesis.go

```diff
@@ -35,7 +35,26 @@ func GetGenesisStateFromAppState(cdc codec.Codec, appState map[string]json.RawMe
 }
 
 func ValidateGenesis(data types.GenesisState) error {
-	return data.Params.Validate()
+	err := data.Params.Validate()
+	if err != nil {
+		return err
+	}
+
+	// Validate contracts
+	for _, addr := range data.ContractAddresses {
```

Comment:
With contracts moving out of the Params (since it was a gov required whitelist before), I wonder if it makes sense to not put contracts or jailed contracts into the genesis file.

instead just putting it in a `store`  and calling it a day. (not a reason to store in genesis anymore is there?)

Reply:
Good point, they would all be jailed instantly regardless! Genesis no longer keeps track of contracts.